internal/driver/config/opl_config_namespace_watcher.go

```diff
@@ -30,15 +35,45 @@ type (
 var _ namespace.Manager = (*oplConfigWatcher)(nil)
 
 func newOPLConfigWatcher(ctx context.Context, l *logrusx.Logger, target string) (*oplConfigWatcher, error) {
-
 	nw := &oplConfigWatcher{
 		logger:                 l,
 		target:                 target,
 		files:                  configFiles{byPath: make(map[string]io.Reader)},
 		memoryNamespaceManager: *NewMemoryNamespaceManager(),
 	}
 
-	return nw, watchTarget(ctx, target, nw, l)
+	targetUrl, err := urlx.Parse(target)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	switch targetUrl.Scheme {
+	case "file":
+		return nw, watchTarget(ctx, target, nw, l)
+	case "http", "https":
+		file, err := download(ctx, targetUrl)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+		nw.files.byPath[targetUrl.String()] = file
+		nw.parseFiles()
+		return nw, err
+	default:
+		return nil, fmt.Errorf("unexpected url scheme: %q", targetUrl.Scheme)
+	}
+}
+
+func download(ctx context.Context, url *url.URL) (io.ReadCloser, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", url.String(), nil)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
If we want to load this in our backoffifce system, you'll need to use a client which is protected against SSRF (see ory kratos / ory hydra private IP ban).